runtime/cpp/include/bpf-api.h

```diff
@@ -64,13 +67,29 @@ class wasm_bpf_program {
 
    public:
     int bpf_map_fd_by_name(const char *name);
-    int load_bpf_object(const void *obj_buf, size_t obj_buf_sz);
+    int load_bpf_object(const void* obj_buf,
+                        size_t obj_buf_sz,
+                        bpf_map_store& maps);
     int attach_bpf_program(const char *name, const char *attach_target);
     int bpf_buffer_poll(wasm_exec_env_t exec_env, int fd, int32_t sample_func,
                         uint32_t ctx, void *buffer_data, size_t max_size,
                         int timeout_ms);
+    void clear_map_fds(bpf_map_store& maps);
+    bpf_map* map_ptr_by_fd(int fd);
 };
 
+/// @brief A user data structure whose instance will be shared in a wasm
+/// runtime. It will store a map containing id->bpf_program and map fds opened
+/// by a bpf program
+/// Note that we need to remove fds opened by a bpf program when it's closed
```

Comment:
>  Note that we need to remove fds opened by a bpf program when it's closed

What fds?

The bpf maps and object fd is closed when bpf object is closed and detached. For attach fds, it should be handled by the WASI runtime and the programs itself.